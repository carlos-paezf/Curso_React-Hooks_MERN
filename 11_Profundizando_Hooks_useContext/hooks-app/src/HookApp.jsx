import React from 'react';
import { CounterApp, CounterWithCustomHook } from './components/01-useState';
import { FormWithCustomHook, SimpleForm } from './components/02-useEffect';
import { MultipleCustomHooks } from './components/03-examples';
import { FocusScreen, RealExampleRef } from './components/04-useRef';
import { LayoutEffect } from './components/05-useLayoutEffect';
import { CallbackHook, MemoHook, Memorize } from './components/06-memos';
import { Padre } from './components/07-tarea-memorize';


export const HookApp = () => {
    return (
        <>
            <h1>Hooks App</h1>
            <hr />

            <table className='table'>
                <tbody>
                    <tr>
                        <th><CounterApp /></th>
                        <th><CounterWithCustomHook /></th>
                    </tr>

                    <tr>
                        <th><SimpleForm /></th>
                        <th><FormWithCustomHook /></th>
                    </tr>

                    <tr>
                        <th colSpan={2}><MultipleCustomHooks /></th>
                    </tr>

                    <tr>
                        <th><FocusScreen /></th>
                        <th><RealExampleRef /></th>
                    </tr>

                    <tr>
                        <th colSpan={2}><LayoutEffect /></th>
                    </tr>

                    <tr>
                        <th><Memorize /></th>
                        <th><MemoHook /></th>
                    </tr>

                    <tr>
                        <th colSpan={2}><CallbackHook /></th>
                    </tr>

                    <tr>
                        <th colSpan={2}><Padre /></th>
                    </tr>
                </tbody>
            </table>
        </>
    )
};
