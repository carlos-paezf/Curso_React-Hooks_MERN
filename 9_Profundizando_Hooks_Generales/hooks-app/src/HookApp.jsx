import React from 'react';
import { CounterApp, CounterWithCustomHook } from './components/01-useState';
import { FormWithCustomHook, SimpleForm } from './components/02-useEffect';
import { MultipleCustomHooks } from './components/03-examples';
import { FocusScreen, RealExampleRef } from './components/04-useRef';


export const HookApp = () => {
    return (
        <>
            <h1>Hooks App</h1>
            <hr />

            <table className='table'>
                <tbody>
                    <tr>
                        <th><CounterApp /></th>
                        <th><CounterWithCustomHook /></th>
                    </tr>

                    <tr>
                        <th><SimpleForm /></th>
                        <th><FormWithCustomHook /></th>
                    </tr>

                    <tr>
                        <th colSpan={2}><MultipleCustomHooks /></th>
                    </tr>

                    <tr>
                        <th><FocusScreen /></th>
                        <th><RealExampleRef /></th>
                    </tr>
                </tbody>
            </table>
        </>
    )
};
