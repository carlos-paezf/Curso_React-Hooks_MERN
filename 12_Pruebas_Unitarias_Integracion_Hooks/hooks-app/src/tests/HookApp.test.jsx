import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { HookApp } from '../HookApp';


describe('Pruebas al componente <HookApp />', () => {
    /* Creating a shallow wrapper for the HookApp component. */
    let wrapper = shallow(<HookApp />)

    /* This is a test setup. It is creating a shallow wrapper for the component. */
    beforeEach(() => {
        wrapper = shallow(<HookApp />)
    })

    /* Testing the component. */
    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })
})