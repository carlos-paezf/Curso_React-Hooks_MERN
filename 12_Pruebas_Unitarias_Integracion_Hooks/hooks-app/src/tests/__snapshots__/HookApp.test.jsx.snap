// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <HookApp /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    Hooks App
  </h1>
  <hr />
  <table
    className="table"
  >
    <tbody>
      <tr>
        <th>
          <CounterApp />
        </th>
        <th>
          <CounterWithCustomHook />
        </th>
      </tr>
      <tr>
        <th>
          <SimpleForm />
        </th>
        <th>
          <FormWithCustomHook />
        </th>
      </tr>
      <tr>
        <th
          colSpan={2}
        >
          <MultipleCustomHooks />
        </th>
      </tr>
      <tr>
        <th>
          <FocusScreen />
        </th>
        <th>
          <RealExampleRef />
        </th>
      </tr>
      <tr>
        <th
          colSpan={2}
        >
          <LayoutEffect />
        </th>
      </tr>
      <tr>
        <th>
          <Memorize />
        </th>
        <th>
          <MemoHook />
        </th>
      </tr>
      <tr>
        <th
          colSpan={2}
        >
          <CallbackHook />
        </th>
      </tr>
      <tr>
        <th
          colSpan={2}
        >
          <Padre />
        </th>
      </tr>
    </tbody>
  </table>
</Fragment>
`;
